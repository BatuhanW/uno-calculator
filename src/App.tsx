import React from "react";
import { useRecoilValue } from "recoil";
import styled from "styled-components";

import { playersState } from "./atoms/states";

import Card from "./components/Card";
import Footer from "./components/Footer";
import Hero from "./components/Hero";

import "bulma/css/bulma.css";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: space-between;
`;

const CardsContainer = styled.div`
  overflow-y: scroll;
  padding-top: 3rem;
  padding-bottom: 15rem;
`;

export interface PlayerPoints {
  key: string;
  playerName: string;
  round: number;
  score: number;
}

function App() {
  const players = useRecoilValue(playersState);
  const [points, setPoints] = React.useState<PlayerPoints[]>([]);

  return (
    <Container>
      <Hero />
      <CardsContainer className="px-5">
        {players.map((player) => (
          <Card
            key={player}
            player={player}
            playerPoints={points.filter(
              (playerPoints) => playerPoints.playerName === player
            )}
            setPoints={setPoints}
          />
        ))}
      </CardsContainer>
      <Footer points={points} />
    </Container>
  );
}

export default App;
