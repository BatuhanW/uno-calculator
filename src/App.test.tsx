import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import { RecoilRoot } from "recoil";

describe("App", () => {
  describe("Snapshot", () => {
    it("should match snapshot", async () => {
      const { container } = render(
        <RecoilRoot>
          <App />
        </RecoilRoot>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
