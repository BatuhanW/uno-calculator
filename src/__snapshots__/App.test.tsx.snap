// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot should match snapshot 1`] = `
<div>
  <div
    class="sc-AxhCb cfrjRJ"
  >
    <section
      class="hero is-primary"
    >
      <div
        class="hero-body"
      >
        <div
          class="container"
        >
          <h1
            class="title"
          >
            UNO Calculator
          </h1>
        </div>
      </div>
    </section>
    <div
      class="sc-AxhUy eECsHo px-5"
    />
    <div
      class="sc-AxiKw fPbTKm is-family-monospace is-size-5"
    >
      <div
        class="content"
      >
        <form>
          <div
            class="field has-addons"
          >
            <div
              class="control"
            >
              <input
                class="input"
                data-cy="newPlayerInput"
                name="playerName"
                placeholder="Add new player"
                type="text"
              />
            </div>
            <div
              class="control"
            >
              <input
                class="button is-primary block"
                data-cy="newPlayerSubmit"
                type="submit"
              />
            </div>
          </div>
        </form>
        <div
          class="my-3"
        />
        <span>
          Player Count: 
          0
        </span>
         |
         
        <button
          class="button is-small is-danger"
        >
          X
        </button>
        <div
          class="my-3"
        />
        <span>
          Round: 
          0
          <span
            class="mx-1"
          />
          <span />
          <span
            class="mx-1"
          />
          <span />
        </span>
        <div
          class="mt-3"
        />
      </div>
    </div>
  </div>
</div>
`;
